src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -2581,6 +2612,9 @@ CheckConnectionReady(WorkerSession *session)
 
 	UpdateConnectionWaitFlags(session, waitFlags);
 
+	/* don't consume input redundantly if we cycle back into CheckConnectionReady */
+	session->latestWaitEvents = 0;
```

Comment:
Very minor, feel free to ignore: The name might be misleading since we're reseting the value right after the first iteration. Maybe `latestUnconsumedWaitEvents`?